listenbrainz/db/timescale.py

```diff
@@ -42,16 +42,25 @@ def run_sql_script_without_transaction(sql_file_path):
         connection = engine.connect()
         connection.connection.set_isolation_level(0)
         lines = sql.read().splitlines()
-        try:
-            for line in lines:
-                # TODO: Not a great way of removing comments. The alternative is to catch
-                # the exception sqlalchemy.exc.ProgrammingError "can't execute an empty query"
-                if line and not line.startswith("--"):
-                    connection.execute(line)
-        except sqlalchemy.exc.ProgrammingError as e:
-            print("Error: {}".format(e))
-            return False
-        finally:
-            connection.connection.set_isolation_level(1)
-            connection.close()
+        while True:
+            try:
+                for line in lines:
+                    # TODO: Not a great way of removing comments. The alternative is to catch
+                    # the exception sqlalchemy.exc.ProgrammingError "can't execute an empty query"
+                    if line and not line.startswith("--"):
+                        connection.execute(line)
+                break
+            except sqlalchemy.exc.ProgrammingError as e:
+                print("Error: {}".format(e))
+                return False
+            except sqlalchemy.exc.OperationalError:
+                print("Trapped template1 access error, FFS! Sleeping, trying again.")
+                retries += 1
+                if retries == 5:
+                    raise
+                sleep(1)
```

Comment:
Need to use as `time.sleep` or modify the import.

Reply:
fixed.